test/functional/cmdLineTests/locales/run_all_locales.sh

```diff
@@ -51,9 +51,13 @@ do
     export LANG=$locale;
     
     # run first argument and pass remaining arguments
-    $1 "${@:2}";
-    
-    if [ "$?" != 0 ]; then
+    output=$($1 "${@:2}" 2>&1) ; result="$?"
+    echo $output
+    grep 'UnsupportedCharsetException' <<< $output 
+    if [ "$?" = 0 ]; then
```

Comment:
These two lines should be combined and grep output suppressed:
```
if grep -q UnsupportedCharsetException <<< "$output" ; then
```
(note also that `UnsupportedCharsetException` need not be quoted and that `$output` should be quoted)